editor/scene_tree_editor.cpp

```diff
@@ -592,13 +567,48 @@ void SceneTreeEditor::_update_tree(bool p_scroll_to_selected) {
 	updating_tree = true;
 	tree->clear();
 	if (get_scene_node()) {
-		_add_nodes(get_scene_node(), nullptr, p_scroll_to_selected);
+		_add_nodes(get_scene_node(), nullptr);
 		last_hash = hash_djb2_one_64(0);
 		_compute_hash(get_scene_node(), last_hash);
 	}
 	updating_tree = false;
-
 	tree_dirty = false;
+
+	if (!filter.is_empty()) {
+		_update_filter(nullptr, p_scroll_to_selected);
+	}
+}
+
+bool SceneTreeEditor::_update_filter(TreeItem *p_parent, bool p_scroll_to_selected) {
+	if (!p_parent) {
+		p_parent = tree->get_root();
+	}
+
+	bool keep = false;
+	for (TreeItem *child = p_parent->get_first_child(); child; child = child->get_next()) {
+		keep = _update_filter(child, p_scroll_to_selected) || keep;
+	}
+
+	if (!keep) {
+		keep = filter.is_subsequence_ofn(p_parent->get_text(0));
```

Comment:
@KoBeWi  I have a question regard this PR's effect on my #58377. After this PR being merged, the p_parent->get_class() will no longer return the type of node, ( it will return something like "Node3D" before this PR).

 I'm actually new to godot's development, do I used a wrong function? Is there any new way to implement #58377 ?
Thank you!

Reply:
Doesn't seem like `p_parent` was ever anything else than TreeItem. Not sure how to get node from a tree item, maybe it's stored in some metadata.

Do you have the old code somewhere?

Reply:
You can get the node pointer by using `get_node(p_parent->get_metadata(0))` and then use `get_class()` on that node.

Reply:
It works, thank you very much!